Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSDetEffCorrVana.py

```diff
@@ -0,0 +1,244 @@
+        arr = np.array(_vana_bg_.extractY()).flatten()
+        neg_values = np.where(arr < 0)[0]
+        if len(neg_values):
+            self._cleanup(wslist)
+            message = "Background " + self.bkgws.getName() + " is higher than Vanadium " + \
+                self.vanaws.getName() + " signal!"
+            self.log().error(message)
+            raise RuntimeError(message)
+
+        # 3. calculate correction coefficients
+        _vana_mean_ws_ = self._vana_mean(_vana_bg_)
+        if not _vana_mean_ws_:
+            self._cleanup(wslist)
+            return None
+        if not self.vana_mean_name:
+            wslist.append(_vana_mean_ws_)
+        _coef_ws_ = api.Divide(LHSWorkspace=_vana_bg_, RHSWorkspace=_vana_mean_ws_, WarnOnZeroDivide=True)
+        wslist.append(_coef_ws_)
+        # 4. correct raw data (not normalized!)
+        api.Divide(LHSWorkspace=self.dataws, RHSWorkspace=_coef_ws_, WarnOnZeroDivide=True,
+                   OutputWorkspace=self.outws_name)
+        outws = api.mtd[self.outws_name]
+        # cleanup
+        self._cleanup(wslist)
+        return outws
+
+    def _check_properties(self, lhs_run, rhs_run):
+        """
+        checks whether properties match
+        """
+        lhs_title = ""
+        rhs_title = ""
+        if lhs_run.hasProperty('run_title'):
+            lhs_title = lhs_run.getProperty('run_title').value
+        if rhs_run.hasProperty('run_title'):
+            rhs_title = rhs_run.getProperty('run_title').value
+
+        for property_name in self.properties_to_compare:
+            if lhs_run.hasProperty(property_name) and rhs_run.hasProperty(property_name):
+                lhs_property = lhs_run.getProperty(property_name)
+                rhs_property = rhs_run.getProperty(property_name)
+                if lhs_property.type == rhs_property.type:
+                    if lhs_property.type == 'string':
+                        if lhs_property.value != rhs_property.value:
+                            message = "Property " + property_name + " does not match! " + \
+                                lhs_title + ": " + lhs_property.value + ", but " + \
+                                rhs_title + ": " + rhs_property.value
+                            self.log().warning(message)
+                    if lhs_property.type == 'number':
+                        if abs(lhs_property.value - rhs_property.value) > 5e-3:
+                            message = "Property " + property_name + " does not match! " + \
+                                lhs_title + ": " + str(lhs_property.value) + ", but " + \
+                                rhs_title + ": " + str(rhs_property.value)
+                            self.log().warning(message)
+                else:
+                    message = "Property " + property_name + " does not match! " + \
+                        lhs_title + ": " + str(lhs_property.value) + ", but " + \
+                        rhs_title + ": " + str(rhs_property.value)
```

Comment:
Maybe here you could add more information saying that it's the types of the properties that do not match, and print also the types.